### PR TITLE
fix(core): fix index miss values or contain duplicates in certain commit patterns

### PR DESCRIPTION
**Pull request by ideoma**
After adding index testing to fuzz tests few issues were found and fixed
- WAL tables can miss a range of rows in the index. When rows were applied from lag, indexes were not incrementally built.
- Non-WAL tables can have duplicated index entries when a row cancel happens on the edge of a partition.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/BitmapIndexWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|

File: core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java [hunk 1]
```diff
@@ -466,6 +466,39 @@ private void applyWalParallel(ObjList<FuzzTransaction> transactions, String tabl
...
+
+                String columnName = metadata.getColumnName(columnIndex);
+                long randomRow = rnd.nextLong(reader.size());
+                sink.clear();
+                TestUtils.printSql(compiler, sqlExecutionContext, "select \"" + columnName + "\" as a from " + tableNameNoWal + " limit " + randomRow + ", 1", sink);
+                String prefix = "a\n";
+                String randomValue = sink.length() > prefix.length() + 2 ? sink.subSequence(prefix.length(), sink.length() - 1).toString() : null;
+                String indexedWhereClause = " where \"" + columnName + "\" = " + (randomValue == null ? "null" : "'" + randomValue + "'");
+                LOG.info().$("checking random index with filter: ").$(indexedWhereClause).I$();
+                String limit = " limit 8970, 9100";
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java:496]: nit: would be nice to add a comment on why we use exactly these limit boundaries here

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java (hunk 1 above)
Comment (ideoma) [reply core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java:496]: ah, that's debugging, removing

Comment (ideoma) [reply core/src/test/java/io/questdb/test/griffin/wal/WalWriterFuzzTest.java]: removed at [a92f9ae](https://github.com/questdb/questdb/pull/3266/commits/a92f9ae47f5978081174d69a3831d8cca0cedf40)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/BitmapIndexWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|

**Pull request merged**